### DISCUSSION TITLE
Foxglove Studio Websocket Over VPN

### DISCUSSION THREAD

**Discussion by aditya2592** [foxglove/community#615] [Q&A] [answered]
Hi, I am trying to use Foxglove Studio to visualize live data streamed by a robot through Foxglove Bridge but I get the error below in foxglove bridge on the robot:

```
[foxglove_bridge-3] [INFO] [1673054996.157762666] [foxglove_bridge]: [WS] Rejecting client 10.253.0.116:43172 which did not declare support for subprotocol foxglove.websocket.v1
[foxglove_bridge-3] [ERROR] [1673054996.158054809] [foxglove_bridge]: [WS] Handshake ended with HTTP error: 400
```

Is there something that needs to be enabled in the VPN settings to allow this data to go through?

**Comment by defunctzombie** [comment, marked as answer]
Which version of foxglove bridge and Studio are you using?

**Reply by aditya2592** [reply]
I am using below for bridge:
`git clone -b 0.2.2 https://github.com/foxglove/ros-foxglove-bridge.git src/foxglove_bridge`

For studio its 1.35.2

**Reply by amacneil** [reply]
Did you definitely select the `Foxglove WebSocket` connection option in Studio, not the `Rosbridge` one?

**Comment by achim-k** [comment]
@aditya2592 can you confirm that Studio sends the correct protocol? You can do that in the chrome developers tools network tab by selecting the websocket connection and inspecting the headers:

![image](https://user-images.githubusercontent.com/9250155/212351963-2e4cc2a2-7264-44f5-b01b-125b02040550.png)
